src/by/epam/lab/Purchase.java

```diff
@@ -0,0 +1,33 @@
+package by.epam.lab;
+
+public class Purchase {
+    private final Priceable product;
+    private final double quantity;
```

Comment:
replace double to Number, because quantity can be both integer or double, it depends